modules/integration/tests-integration/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/header/ESBJAVA2283ReturnContentTypeTestCase.java

```diff
@@ -18,52 +18,80 @@
+import com.sun.net.httpserver.HttpServer;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import org.apache.axiom.om.OMElement;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.DefaultHttpClient;
+import static org.testng.Assert.assertEquals;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
-import org.wso2.carbon.integration.common.utils.ClientConnectionUtil;
-import org.wso2.carbon.automation.extensions.servers.httpserver.SimpleHttpClient;
-import org.wso2.carbon.automation.extensions.servers.webserver.SimpleWebServer;
 import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
-
-import java.net.InetAddress;
-import java.util.HashMap;
-import java.util.Map;
-
-import static org.testng.Assert.assertEquals;
+import org.wso2.esb.integration.common.utils.ESBTestCaseUtils;
 
 public class ESBJAVA2283ReturnContentTypeTestCase extends ESBIntegrationTest {
     private Log log = LogFactory.getLog(ESBJAVA2283ReturnContentTypeTestCase.class);
+	private HttpServer server = null;
 
     @BeforeClass(alwaysRun = true)
     public void init() throws Exception {
-        super.init();
-        loadESBConfigurationFromClasspath("/artifacts/ESB/synapseconfig/esbjava2283/synapse.xml");
+	    super.init();
+	    String relativePath = "/artifacts/ESB/synapseconfig/esbjava2283/api.xml";
+	    ESBTestCaseUtils util = new ESBTestCaseUtils();
+	    relativePath = relativePath.replaceAll("[\\\\/]", File.separator);
+	    OMElement apiConfig = util.loadResource(relativePath);
+	    addApi(apiConfig);
     }
 
     @Test(groups = {"wso2.esb"}, description = "test return content type")
     public void testReturnContentType() throws Exception {
-        SimpleWebServer simpleWebServer = new SimpleWebServer(9005, 200);
-        try {
-            simpleWebServer.start();
-            SimpleHttpClient httpClient = new SimpleHttpClient();
-            Map<String, String> headers = new HashMap<String, String>();
-            headers.put("Content-Type", "text/xml");
-            HttpResponse response = httpClient.doGet(context.getContextUrls().getServiceUrl() + "/passThoughProxy", headers);
-            log.info(response.getEntity().getContentType());
-            log.info(response.getStatusLine().getStatusCode());
+	    int port = 8089;
+	    String contentType = "text/xml";
+	    server = HttpServer.create(new InetSocketAddress(port), 0);
+	    server.createContext("/gettest", new MyHandler());
+	    server.setExecutor(null); // creates a default executor
+	    server.start();
+	    DefaultHttpClient httpclient = new DefaultHttpClient();
+	    String url = "http://localhost:8280/serviceTest/test";
+	    HttpGet httpGet = new HttpGet(url);
```

Comment:
Thanks a lot for pointing this out. Issue is fixed with PR https://github.com/wso2/product-esb/pull/368
Since product esb uses httpclient 4.1.2, it doesn't have HttpGet.releaseConnection() method. Accoding to [1] to ensure proper release of system resources one must close either the content stream associated with the entity or the response itself.
[1]. https://hc.apache.org/httpcomponents-client-ga/tutorial/html/fundamentals.html#d5e145